### PR TITLE
ref: migrate public-facing submission types to shared folder

### PR DESCRIPTION
**Pull request by karrui**
## Problem

This PR moves public-facing submission types to the shared root folder and extends them in the client and server.

See #2066

## Solution

**Breaking Changes** 

- [ ] Yes - this PR contains breaking changes
    - Details ...
- [x] No - this PR is backwards compatible  
  - Only moving types

**Features**:

- ref: add and use public storage mode submission types

**Improvements**:

- feat: remove unused ClientSubmission interface
- clean up now unused response types

### CODE REVIEW

File: shared/types/response.ts [hunk 1]
```diff
@@ -1,7 +1,10 @@
+import { Opaque } from 'type-fest'
 import { z } from 'zod'
 import { BasicField, MyInfoAttribute } from './field'
 
 const ResponseBase = z.object({
+  myInfo: z.never().optional(),
+  signature: z.never().optional(),
```
Comment (mantariksh) [inline shared/types/response.ts:7]: just making sure - does this work when merging with `MyInfoResponseBase` and `VerifiableResponseBase`?

File: shared/types/submission.ts [hunk 2]
```diff
@@ -87,3 +94,31 @@ export type FormSubmissionMetadataQueryDto = RequireAtLeastOne<
...
 >
+
+type SubmissionContentBase = {
+  responses: FieldResponse[]
+}
+
+/**
+ * Shape of email form submissions
+ */
+export type EmailModeSubmissionContentDto = SubmissionContentBase
```
Comment (mantariksh) [inline shared/types/submission.ts:105]: tiny nit but this is really a mouthful - any issue with the old `EmailSubmissionDto` naming?

File: src/types/api/email_submission.ts [hunk 3]
```diff
@@ -1,5 +1,28 @@
...
-export type EmailSubmissionDto = {
-  responses: ({ question: string } & FieldResponse)[]
+import {
+  AttachmentResponse,
+  FieldResponse,
+} from '../../../shared/types/response'
+import { EmailModeSubmissionContentDto } from '../../../shared/types/submission'
+
+/**
+ * AttachmentResponses with additional server injected metadata on email forms..
```
Comment (mantariksh) [inline src/types/api/email_submission.ts:10]: two full stops means twice as final issit

**Review by mantariksh — APPROVED** [review]

File: shared/types/response.ts (hunk 1 above)
Comment (karrui) [reply shared/types/response.ts:7]: Yea checked, the base type only accepts undefined, but extended types accept the correct items

File: shared/types/submission.ts (hunk 2 above)
Comment (karrui) [reply shared/types/submission.ts:105]: EmailSubmissionDto gives the wrong impression that that is the entire shape of the DTO, when in actuality the content has to be transformed into a `FormData` object prior to calling the endpoint. Just thought to make that distinction. Your thoughts?

Comment (karrui) [reply shared/types/submission.ts:105]: Will fix in another PR if you think we should still change :P

**Pull request merged**